Lib/test/test_pprint.py

```diff
@@ -533,9 +533,14 @@ def test_dataclass_with_repr(self):
         self.assertEqual(formatted, "custom repr that doesn't fit within pprint width")
 
     def test_dataclass_no_repr(self):
+        import re
+
         dc = dataclass3()
         formatted = pprint.pformat(dc, width=10)
-        self.assertRegex(formatted, r"<test.test_pprint.dataclass3 object at \w+>")
+        self.assertRegex(
+            formatted,
+            re.escape(f"<{__name__}.dataclass3 object at ") + r"\w+>",
+        )
```

Comment:
I think it would look clearer if only escape `__name__`.
```suggestion
        self.assertRegex(formatted,
                         fr"<{re.escape(__name__)}.dataclass3 object at \w+>")
```
But it is only a nitpick.